pybutton/request.py

```diff
@@ -50,21 +51,31 @@ def request(url, method, headers, data=None):
         if data:
             request.add_header('Content-Type', 'application/json')
 
-        response = urlopen(request).read().decode('utf8')
+        response = urlopen(request, timeout=timeout).read().decode('utf8')
 
         try:
             return json.loads(response)
         except ValueError:
             raise ButtonClientError('Invalid response: {0}'.format(response))
 
-    __all__ = [Request, urlopen, HTTPError, request]
+    def request_url(secure, hostname, port, path):
```

Comment:
because this is the exact same implementation as the one on line 117, we can simply import each `urlunsplit` in the `if/else` and the define `request_url` and `__all__` at the bottom of the file, once.  Also, would you mind adding a docstring on this because it's used external to this file?